### PR TITLE
PMM-7847 API for MySQL TLS.

### PR DESCRIPTION
**Pull request by JiriCtvrtka**
[PMM-7847](https://jira.percona.com/browse/PMM-7847)

- [x] Build: https://github.com/Percona-Lab/pmm-submodules/pull/1689

- [x] Links to other linked pull requests (optional).
pmm: https://github.com/percona/pmm/pull/720
pmm-agent: https://github.com/percona/pmm-agent/pull/249
pmm-admin: https://github.com/percona/pmm-admin/pull/145
---
- [x] Tests passed.
- [ ] Feature build pass.
- [x] (Re)requested review.
- [x] Fix conflicts with target branch.
- [ ] Update API dependency.
- [x] Update jira ticket description if needed.
- [x] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

**Review by askomorokhov — APPROVED** [review]

**Review by percona-csalguero — APPROVED** [review]

**Review by askomorokhov — APPROVED** [review]

File: services/management/grpc/actions_server.go [hunk 1]
```diff
@@ -131,12 +131,12 @@ func (s *actionsServer) prepareServiceActionWithFiles(serviceID, pmmAgentID, dat
 // StartMySQLExplainAction starts MySQL EXPLAIN Action with traditional output.
 //nolint:lll
 func (s *actionsServer) StartMySQLExplainAction(ctx context.Context, req *managementpb.StartMySQLExplainActionRequest) (*managementpb.StartMySQLExplainActionResponse, error) {
-	res, dsn, err := s.prepareServiceAction(req.ServiceId, req.PmmAgentId, req.Database)
+	res, dsn, files, tdp, err := s.prepareServiceActionWithFiles(req.ServiceId, req.PmmAgentId, "admin")
```
Comment (BupycHuk) [inline services/management/grpc/actions_server.go:134]:
```suggestion
	res, dsn, files, tdp, err := s.prepareServiceActionWithFiles(req.ServiceId, req.PmmAgentId, req.Database)
```
here and everywhere below

File: services/agents/mysql_test.go [hunk 2]
```diff
@@ -158,10 +182,16 @@ func TestMySQLdExporterConfigTablestatsGroupDisabled(t *testing.T) {
 			"--web.listen-address=:{{ .listen_port }}",
 		},
```
Comment (BupycHuk) [inline services/agents/mysql_test.go:183]: let's add to this test check that we have new args `mysql.ssl-cert-file` and `mysql.ssl-key-file`

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: services/management/grpc/actions_server.go (hunk 1 above)
Comment (JiriCtvrtka) [reply services/management/grpc/actions_server.go]: Thank you.

File: services/agents/mysql_test.go (hunk 2 above)
Comment (JiriCtvrtka) [reply services/agents/mysql_test.go:185]: Added.

File: services/agents/mysql_test.go [hunk 3]
```diff
@@ -155,13 +179,20 @@ func TestMySQLdExporterConfigTablestatsGroupDisabled(t *testing.T) {
 			"--exporter.global-conn-pool",
 			"--exporter.max-idle-conns=3",
 			"--exporter.max-open-conns=3",
+			"--mysql.ssl-cert-file={{ .TextFiles.tlsCert }}",
+			"--mysql.ssl-key-file={{ .TextFiles.tlsKey }}",
 			"--web.listen-address=:{{ .listen_port }}",
```
Comment (BupycHuk) [inline services/agents/mysql_test.go:184]: looks like `mysql.ssl-ca-file` is missed.

File: models/agent_model.go [hunk 4]
```diff
@@ -262,13 +276,7 @@ func (s *Agent) DSN(service *Service, dialTimeout time.Duration, database string
...
 		cfg.Params = make(map[string]string)
 		if s.TLS {
-			// TODO: how certs and other parameters are going to be specified? We need to implement calling RegisterTLSConfig
-			// See https://godoc.org/github.com/go-sql-driver/mysql#RegisterTLSConfig
-			if s.TLSSkipVerify {
-				cfg.Params["tls"] = "skip-verify"
-			} else {
-				cfg.Params["tls"] = "true"
-			}
+			cfg.Params["tls"] = "custom"
```
Comment (BupycHuk) [inline models/agent_model.go:279]: I would say that custom should be if custom certificates are not empty. if skip-verify is passed it should be `skip-verify` and if just `tls` is passed then it should be `true`.

Comment (BupycHuk) [reply models/agent_model.go:279]: let's include it to the new task.

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**